oidc-authenticator/src/main/java/org/xwiki/contrib/oidc/auth/internal/endpoint/CallbackOIDCEndpoint.java

```diff
@@ -235,6 +241,35 @@ public Response handle(HTTPRequest httpRequest, OIDCResourceReference reference)
             // TODO: add support for null ClientProvider
             idToken = new IDTokenClaimsSet(tokenResponse.getOIDCTokens().getIDToken().getJWTClaimsSet());
         }
+        
+        // Check if ACR is specified and if yes, if value from config matches value returned in id token
+        JSONObject claimsJsonObject = this.configuration.getClaimsRequest().toJSONObject();
```

Comment:
I find a bit surprising to serialize the object to json to check values in it, you can directly use things like this.configuration.getClaimsRequest().getIDTokenClaimsRequest() and ClaimsSetRequest#get("acr") for example.

I would have expected the IDTokenValidator to deal with stuff like this, but while I see it support a concept of nonce, it's not the "acr".

Reply:
I guess it also means that we have to add support for "acr" in the provider (it does not do anything about it right now). But it could be another issue to not forget, there is no reason to block this authenticator focused PR just for this.

Reply:
> I find a bit surprising to serialize the object to json to check values in it, you can directly use things like this.configuration.getClaimsRequest().getIDTokenClaimsRequest() and ClaimsSetRequest#get("acr") for example.

Hm, I tried to find an easier alternative but apparently overlooked this obvious one. I will adapt it once I have time today or on monday.

> I would have expected the IDTokenValidator to deal with stuff like this, but while I see it support a concept of nonce, it's not the "acr".

Yes, that was also what I had hoped :)

> I guess it also means that we have to add support for "acr" in the provider (it does not do anything about it right now). But it could be another issue to not forget, there is no reason to block this authenticator focused PR just for this (even if it would allow writing an integration test for it :)).

If you want to support step-up, then yes. AFAIK usage is not very widespread yet, but it has a lot of potential. Since we don't use XWiki as our IdP, adding ACR handling to the provider is out of scope for us.

Reply:
Modified as suggested and also updated the unit test to use the provided classes